Report originally resolved module as part of reasons

smikula opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->
<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->


<!-- In addition to that please answer these questions: -->

**What kind of change does this PR introduce?**

This addresses #8606.  Currently when modules are scope hoisted there is not enough information to reconstruct the original module dependency graph.  With this change the `reasons` structure in the reported stats contains a field for the originally resolved module.

<!-- E.g. a bugfix, feature, refactoring, build related change, etc… -->

**Did you add tests for your changes?**

Not yet, but I'm happy to if needed.  Currently not all tests seem to be passing, possible because this is the alpha branch.

<!-- Note that we won't merge your changes if you don't add tests -->

**Does this PR introduce a breaking change?**

No, this should be purely additive.

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**What needs to be documented once your changes are merged?**

The new property in the `reason` structure.
<!-- List all the information that needs to be added to the documentation after merge -->
<!-- When your changes are merged you will be asked to contribute this to the documentation -->

sokra commented on lib/stats/DefaultStatsFactoryPlugin.js:
```diff
@@ -509,6 +509,9 @@ const SIMPLE_EXTRACTORS = {
 				moduleName: reason.originModule
 					? reason.originModule.readableIdentifier(requestShortener)
 					: null,
+				resolvedModule: reason.resolvedOriginModule
```
resolvedModuleId and resolvedModuleIdentifier too

smikula commented:
@sokra: I added this to the `toString` output as an optional last field in the reason line -- optional in that it only shows up if the resolved module is different from the actual module.  I've also added a test and updated the snapshots.

smikula commented:
I'm not clear why the test case is failing only for Node 10 on Windows.  It looks like in that particular case the reasons are reported in a different order so they don't match the snapshot.  Is there anything that would be non-deterministic about this ordering?

sokra commented on lib/stats/DefaultStatsPrinterPlugin.js:
```diff
@@ -510,7 +511,8 @@ const PREFERED_ORDERS = {
 		"moduleId",
 		"module",
 		"loc",
```
Place it between `module` and `loc`

smikula commented:
@sokra, thanks for the pointers.  I've made the changed you suggested and it looks like that took care of the test failure.

smikula commented:
Thanks for the approval.  Is there anything else I need to do before you merge it?
